spec/services/preserved_object_handler_update_version_spec.rb

```diff
@@ -340,4 +334,370 @@
+          allow(pc).to receive(:save!)
+          allow(po_handler).to receive(:moab_validation_errors).and_return(['foo'])
+          po_handler.update_version_after_validation
+          expect(po).to have_received(:save!)
+          expect(pc).to have_received(:save!)
+        end
+
+        it 'does not call PreservedObject.touch and PreservedCopy.touch if the existing record is NOT altered' do
+          po_handler = described_class.new(druid, 1, 1, ep)
+          po = instance_double(PreservedObject)
+          allow(PreservedObject).to receive(:find_by).with(druid: druid).and_return(po)
+          allow(po).to receive(:current_version).and_return(1)
+          allow(po).to receive(:changed?).and_return(false)
+          allow(po).to receive(:touch)
+          pc = instance_double(PreservedCopy)
+          allow(PreservedCopy).to receive(:find_by).with(preserved_object: po, endpoint: ep).and_return(pc)
+          allow(pc).to receive(:version).and_return(1)
+          allow(pc).to receive(:endpoint).with(ep)
+          allow(pc).to receive(:status).and_return(instance_double(Status, status_text: 'ok'))
+          allow(pc).to receive(:status=)
+          allow(pc).to receive(:last_audited=)
+          allow(pc).to receive(:last_checked_on_storage=)
+          allow(pc).to receive(:changed?).and_return(false)
+          allow(pc).to receive(:touch)
+          po_handler.update_version_after_validation
+          expect(po).not_to have_received(:touch)
+          expect(pc).not_to have_received(:touch)
+        end
+
+        context 'incoming version newer than catalog versions (both) (happy path)' do
+          it 'updates PreservedCopy last_audited' do
+            orig_timestamp = pc.last_audited
+            sleep 1
+            po_handler.update_version_after_validation
+            expect(pc.reload.last_audited).to be > orig_timestamp
+          end
+          it 'updates PreservedCopy last_checked_on_storage' do
+            orig_timestamp = pc.last_checked_on_storage
+            po_handler.update_version_after_validation
+            expect(pc.reload.last_checked_on_storage).to be > orig_timestamp
+          end
+
+          it 'calls #update_online_version with validated = true and Status.invalid_moab' do
+            expect(po_handler).to receive(:update_online_version).with(true, Status.invalid_moab).and_call_original
+            po_handler.update_version_after_validation
+            skip 'test is lame b/c we only indirectly show the effects of #update_online_version in #update_version specs'
+          end
+
+          it 'updates PreservedCopy status to "moab_invalid" if it was "ok"' do
+            pc.status = Status.ok
+            pc.save!
+            po_handler.update_version_after_validation
+            expect(pc.reload.status).to eq Status.invalid_moab
+          end
+        end
+
+        RSpec.shared_examples 'unexpected version with validation' do |incoming_version|
```

Comment:
can this be moved to the `shared_examples_preserved_object_handler.rb` ?

Reply:
It is only useful in this one context, so I left it here.  I know these tests are long;  I went with "shameless green".  Happy to entertain ideas for shortening them, etc.